### PR TITLE
Lint and refact

### PR DESCRIPTION
**Pull request by fredbi**
Major refactoring of expander

**This PR is mostly refactoring and applies only minor change on behavior**
## Content

* Split up expander and **separate swagger specific feature from schema generics (e.g. resolve ref)**
* Remove duplicate code with parameter / response resolve/expand
* Modernize code style and abide by golangci linters
* Update bindata generation with kevinburke's fork
* Augment UT coverage

## Fixups
Minor bug fixes:
* fixed Parameter.RemoveParam() func
* fixed non deterministic func Operation.SuccessReponse()

## Integration tests
* [x] Integration tests with go-openapi/validate
* [x] Integration tests w/ go-swagger/go-swagger

### CODE REVIEW

**Comment by fredbi** [comment]
Reported golangci failure: golangci/golangci#25

**Comment by fredbi** [comment]
Integration testing went okay for me

File: operation.go [hunk 1]
```diff
@@ -84,11 +89,17 @@ func (o *Operation) SuccessResponse() (*Response, int, bool) {
 		return nil, 0, false
 	}
 
-	for k, v := range o.Responses.StatusCodeResponses {
-		if k/100 == 2 {
-			return &v, k, true
+	responseCodes := make([]int, 0, len(o.Responses.StatusCodeResponses))
```
Comment (casualjim) [inline operation.go]: Why do you replace this with a more inefficient way of doing it?
the `k/100 == 2` has the exact same effect

Comment (fredbi) [reply operation.go]: More inefficient? Mmh interesting point of view. It sure is shorter, but mine gives a clearer intent. About _efficiency_, I am not so sure. Let's get the generated assembly to figure out. Merry Christmas.

Comment (fredbi) [reply operation.go]: My version is slightly faster on a x86, by a few nanosecs :)
See [benchmark](https://gist.github.com/fredbi/208f0b95b57ba5763ceb1a482f6212ec).

All this just to say that this is mostly a matter of taste and mind mapping (you think "2xx", I think [200,300[) more than _efficiency_.

The generated assembly (go11, x86 64bit) is a little bit shorter with the `&&` version, with **FIVE** (!!) assembly ops less... Readers will surely take notice of the excellent optimization job carried out with golang here.

Cheers,

F.

Ex:
`return k/100 == 2` 
yields:
```asm
	MOVQ	$-6640827866535438581, AX
	MOVQ	"".k+8(SP), CX
	IMULQ	CX
	ADDQ	CX, DX
	SARQ	$6, DX
	SARQ	$63, CX
	SUBQ	CX, DX
	CMPQ	DX, $2
	SETEQ	"".~r1+16(SP)
	RET
```
Whereas:
`return k>=200 && k<300` 
yields:
```asm
	MOVQ	"".k+8(SP), AX
	ADDQ	$-200, AX
	CMPQ	AX, $100
	SETCS	"".~r1+16(SP)
	RET
```

Comment (fredbi) [reply operation.go]: Ah ok, I may explain this (and I should have documented this change).

Well, it is simply that in the case of multiple success responses (like '200', '201', '203'...), the previous version did
randomly return one of these.

The additional complexity is here to ensure that this function is deterministic. Hence the append and sort.

File: operation.go [hunk 2]
```diff
@@ -224,7 +232,7 @@ func (o *Operation) AddParam(param *Parameter) *Operation {
 // RemoveParam removes a parameter from the operation
 func (o *Operation) RemoveParam(name, in string) *Operation {
 	for i, p := range o.Parameters {
-		if p.Name == name && p.In == name {
+		if p.Name == name && p.In == in {
```
Comment (gregmarr) [inline operation.go]: PR description says no behavior changes, but this looks like a behavior change.  Does this need to be noted?

Comment (fredbi) [reply operation.go]: Yes you're right: I forgot to mention it. I added such minor changes while adding more tests (which did not work as expected, etc..)

Comment (fredbi) [reply operation.go]: Done

**Comment by fredbi** [comment]
What do we do with this one? I made it on purpose to be able to continue maintenance from a reduced debt. Anything left I should change?

**Pull request merged**